pkg/persistence/object_store_test.go

```diff
@@ -541,6 +542,134 @@ func TestGetDownloadURL(t *testing.T) {
+// returned.
+func TestNewObjectBackupStore(t *testing.T) {
+	tests := []struct {
+		name              string
+		location          *velerov1api.BackupStorageLocation
+		objectStoreGetter objectStoreGetter
+		wantBucket        string
+		wantPrefix        string
+		wantErr           string
+	}{
+		{
+			name:     "location with no ObjectStorage field results in an error",
+			location: &velerov1api.BackupStorageLocation{},
+			wantErr:  "backup storage location does not use object storage",
+		},
+		{
+			name: "location with no Provider field results in an error",
+			location: &velerov1api.BackupStorageLocation{
+				Spec: velerov1api.BackupStorageLocationSpec{
+					StorageType: velerov1api.StorageType{
+						ObjectStorage: new(velerov1api.ObjectStorageLocation),
+					},
+				},
+			},
+			wantErr: "object storage provider name must not be empty",
+		},
+		{
+			name: "location with a Bucket field with a '/' in the middle results in an error",
+			location: &velerov1api.BackupStorageLocation{
+				Spec: velerov1api.BackupStorageLocationSpec{
+					Provider: "provider-1",
+					StorageType: velerov1api.StorageType{
+						ObjectStorage: &velerov1api.ObjectStorageLocation{
+							Bucket: "invalid/bucket",
+						},
+					},
+				},
+			},
+			wantErr: "backup storage location's bucket name \"invalid/bucket\" must not contain a '/' (if using a prefix, put it in the 'Prefix' field instead)",
+		},
+		{
+			name: "when Bucket has a leading and trailing slash, they are both stripped",
+			location: &velerov1api.BackupStorageLocation{
+				Spec: velerov1api.BackupStorageLocationSpec{
+					Provider: "provider-1",
+					StorageType: velerov1api.StorageType{
+						ObjectStorage: &velerov1api.ObjectStorageLocation{
+							Bucket: "/bucket/",
+						},
+					},
+				},
+			},
+			objectStoreGetter: objectStoreGetter{
+				"provider-1": cloudprovider.NewInMemoryObjectStore("bucket"),
+			},
+			wantBucket: "bucket",
+		},
+		{
+			name: "when Prefix has a leading and trailing slash, the leading slash is stripped and the trailing slash is left",
+			location: &velerov1api.BackupStorageLocation{
+				Spec: velerov1api.BackupStorageLocationSpec{
+					Provider: "provider-1",
+					StorageType: velerov1api.StorageType{
+						ObjectStorage: &velerov1api.ObjectStorageLocation{
+							Bucket: "bucket",
+							Prefix: "/prefix/",
+						},
+					},
+				},
+			},
+			objectStoreGetter: objectStoreGetter{
+				"provider-1": cloudprovider.NewInMemoryObjectStore("bucket"),
+			},
+			wantBucket: "bucket",
+			wantPrefix: "prefix/",
+		},
+		{
+			name: "when Prefix has no leading or trailing slash, a trailing slash is added",
+			location: &velerov1api.BackupStorageLocation{
+				Spec: velerov1api.BackupStorageLocationSpec{
+					Provider: "provider-1",
+					StorageType: velerov1api.StorageType{
+						ObjectStorage: &velerov1api.ObjectStorageLocation{
+							Bucket: "bucket",
+							Prefix: "/prefix/",
```

Comment:
I think `Prefix: "/prefix/"` needs to be `Prefix: "prefix"`?

Reply:
yes, good catch! fixed.